src/main/java/core/basesyntax/model/dao/GenericDao.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.model.dao;
+
+import java.util.List;
+
+public interface GenericDao<T> {
```

Comment:
although overall it is a good idea to create GenericDao, for this task you will have trouble with field injection

let's remove this and just have two separate interfaces